### ISSUE TITLE
3.19 Connection issue

### ISSUE THREAD

**Issue by dbzix** [NordSecurity/nordvpn-linux#639]
Having 2 Linux Mint machines with the same configuration, the first one works perfectly, but the second couldn't connect.

Failure log:
```
nordvpnd[426679]: 2024/10/03 16:09:06 [Info] starting vpn
nordvpnd[426679]: 2024/10/03 16:09:06 [Info] libtelio version: v5.0.0
nordvpnd[426679]: 2024/10/03 16:09:06 [Info] TELIO(v5.0.0): "telio::ffi":380 Telio::start entry with instance id: ********************. Public key: "***********". Adapter: LinuxNativeTun. Name: nordlynx
nordvpnd[426679]: task started - Wg
nordvpnd[426679]: 2024-10-03T13:09:06.758136Z ERROR libmoose::tracker::tracker: ContextNotFound Error while fetching shared context: ContextNotFound
nordvpnd[426679]: task started - Nurse QoS Analytics
nordvpnd[426679]: task started - Nurse Heartbeat Analytics
nordvpnd[426679]: task started - Nurse
nordvpnd[426679]: task started - Telio
nordvpnd[426679]: 2024/10/03 16:09:06 [Info] TELIO(v5.0.0): "telio::ffi":499 Telio::set_fwmark entry with instance id: ********************. fwmark: 57841
csc_vpnagent[144804]: IP addresses from active interfaces: nordlynx: 10.5.0.2 wlo1: **************, *****************************
nordvpnd[426679]: 2024/10/03 16:09:06 [Info] TELIO(v5.0.0): "telio::ffi":561 Telio::connect_to_exit_node entry with instance id :********************. Public Key: "WcmF...jUM=". Allowed IP: Some([IP(****************)/0]). Endpoint: Some(IP(****************):51820)
nordvpnd[426679]: 2024/10/03 16:09:06 [Info] TELIO(v5.0.0): "telio::ffi":588 Telio::connect_to_exit_node_with_id entry with instance id :********************. Identifier: None, Public Key: "***********". Allowed IP: Some([IP(****************)/0]). Endpoint: Some(IP(****************):51820)
nordvpnd[426679]: 2024/10/03 16:09:06 [Info] TELIO(v5.0.0): "telio::device::wg_controller":226 Inserting peer: RequestedPeer { peer: Peer { public_key: "***********", endpoint: Some(IP(****************):51820), endpoint_changed_at: None, ip_addresses: [IP(****************), IP(****************)], persistent_keepalive_interval: Some(25), allowed_ips: [IP(****************)/0], rx_bytes: None, time_since_last_rx: None, tx_bytes: None, time_since_last_handshake: None, preshared_key: None }, endpoint: None }
nordvpnd[426679]: 2024/10/03 16:09:06 [Info] received event telio.EventNode: {"Body":{"Identifier":"************************************","PublicKey":"***","Nickname":null,"State":2,"LinkState":1,"IsExit":true,"IsVpn":true,"IpAddresses":["10.5.0.1","100.64.0.1"],"AllowedIps":["0.0.0.0/0"],"Endpoint":"37.19.218.143:51820","Hostname":null,"AllowIncomingConnections":false,"AllowPeerSendFiles":false,"Path":2,"AllowMulticast":false,"PeerAllowsMulticast":false}}
nordvpnd[426679]: 2024/10/03 16:09:06 [Debug] notifying about connect event: {IsMeshnetPeer:false APIHostname: Auto:false ThreatProtectionLite:false Protocol:UNKNOWN_PROTOCOL ResponseServersCount:0 ResponseTime:0 DurationMs:0 ServerFromAPI:false EventStatus:0 TargetServerSelection: Technology:UNKNOWN_TECHNOLOGY TargetServerCity:Kyiv TargetServerCountry:Ukraine TargetServerDomain:ua52.nordvpn.com TargetServerGroup: TargetServerIP:37.19.218.143 TargetServerPick: TargetServerPickerResponse: TargetServerName:Ukraine #52 Error:<nil>}
nordvpnd[426679]: 2024/10/03 16:09:06 [Warning] TELIO(v5.0.0): "telio_pmtu":88 Failed to probe PMTU: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }
NetworkManager[1373]: <info>  [1727960946.8189] device (nordlynx): state change: unmanaged -> unavailable (reason 'connection-assumed', sys-iface-state: 'external')
NetworkManager[1373]: <info>  [1727960946.8194] device (nordlynx): state change: unavailable -> disconnected (reason 'connection-assumed', sys-iface-state: 'external')
NetworkManager[1373]: <info>  [1727960946.8203] device (nordlynx): Activation: starting connection 'nordlynx' (************************************)
NetworkManager[1373]: <info>  [1727960946.8223] device (nordlynx): state change: disconnected -> prepare (reason 'none', sys-iface-state: 'external')
NetworkManager[1373]: <info>  [1727960946.8225] device (nordlynx): state change: prepare -> config (reason 'none', sys-iface-state: 'external')
NetworkManager[1373]: <info>  [1727960946.8227] device (nordlynx): state change: config -> ip-config (reason 'none', sys-iface-state: 'external')
NetworkManager[1373]: <info>  [1727960946.8228] device (nordlynx): state change: ip-config -> ip-check (reason 'none', sys-iface-state: 'external')
NetworkManager[1373]: <info>  [1727960946.8528] device (nordlynx): state change: ip-check -> secondaries (reason 'none', sys-iface-state: 'external')
NetworkManager[1373]: <info>  [1727960946.8530] device (nordlynx): state change: secondaries -> activated (reason 'none', sys-iface-state: 'external')
NetworkManager[1373]: <info>  [1727960946.8536] device (nordlynx): Activation: successful, device activated.
csc_vpnagent[144804]: Function: GetDNSConfig File: ../../vpn/Common/Utility/linux/DBusNMHelper.cpp Line: 295 Unable to get any DNS server for interface nordlynx
csc_vpnagent[144804]: Function: GetDNSConfig File: ../../vpn/Common/Utility/linux/DBusNMHelper.cpp Line: 295 Unable to get any DNS server for interface nordlynx
nordvpnd[426679]: 2024/10/03 16:09:36 [Debug] notifying about disconnect event
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] TELIO(v5.0.0): "telio::ffi":706 Telio::disconnect_from_exit_nodes entry with instance id: ********************.
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] TELIO(v5.0.0): "telio::device::wg_controller":209 Removing peer: Peer { public_key: "***********", endpoint: Some(IP(****************):51820), endpoint_changed_at: Some(Instant { tv_sec: 82554, tv_nsec: 762079487 }), ip_addresses: [IP(****************), IP(****************)], persistent_keepalive_interval: Some(25), allowed_ips: [IP(****************)/0], rx_bytes: Some(0), time_since_last_rx: None, tx_bytes: Some(888), time_since_last_handshake: None, preshared_key: None }
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] TELIO(v5.0.0): "telio::ffi":706 Telio::disconnect_from_exit_nodes entry with instance id: ********************.
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] TELIO(v5.0.0): "telio::ffi":445 Telio::stop entry with instance id: ********************.
nordvpnd[426679]: task stopped - Telio
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] received event telio.EventNode: {"Body":{"Identifier":"************************************","PublicKey":"***","Nickname":null,"State":1,"LinkState":1,"IsExit":true,"IsVpn":true,"IpAddresses":["10.5.0.1","100.64.0.1"],"AllowedIps":["0.0.0.0/0"],"Endpoint":"37.19.218.143:51820","Hostname":null,"AllowIncomingConnections":false,"AllowPeerSendFiles":false,"Path":2,"AllowMulticast":false,"PeerAllowsMulticast":false}}
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] TELIO(v5.0.0): "telio_nurse::nurse":251 Attempting to send moose disconnect event ...
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] TELIO(v5.0.0): "telio_nurse::nurse":294 Moose disconnect event result: Ok(Success)
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] TELIO(v5.0.0): "telio_nurse::nurse":378 Flushing moose changes result: Ok(Success)
nordvpnd[426679]: task stopped - Nurse
nordvpnd[426679]: task stopped - Nurse Heartbeat Analytics
nordvpnd[426679]: task stopped - Nurse QoS Analytics
nordvpnd[426679]: task stopped - Wg
NetworkManager[1373]: <info>  [1727960976.4079] device (nordlynx): state change: activated -> unmanaged (reason 'unmanaged', sys-iface-state: 'removed')
nordvpnd[426679]: 2024/10/03 16:09:36 [Warning] removing suppress prefix rule: no such file or directory
nordvpnd[426679]: 2024/10/03 16:09:36 [Warning] removing fwmark rule: no such file or directory
nordvpnd[426679]: 2024/10/03 16:09:36 [Warning] removing suppress prefix rule: no such file or directory
nordvpnd[426679]: 2024/10/03 16:09:36 [Warning] removing fwmark rule: no such file or directory
nordvpnd[426679]: 2024/10/03 16:09:36 [Defer] unsetting ifgroups: not set
nordvpnd[426679]: 2024/10/03 16:09:36 [Info] TELIO(v5.0.0): "telio::ffi":706 Telio::disconnect_from_exit_nodes entry with instance id: ********************.
nordvpnd[426679]: 2024/10/03 16:09:36 [Error] TELIO(v5.0.0): "telio::ffi":889 Telio::disconnect_from_exit_nodes: Err(NotStarted)
nordvpnd[426679]: 2024/10/03 16:09:36 [Defer] disconnecting from libtelio: stopping libtelio: TelioError: UnknownError: Inner=NotStarted
nordvpnd[426679]: 2024/10/03 16:09:36 [Error] failed to connect to ua52.nordvpn.com : context deadline exceeded
```

**Comment by mariusSincovici** [comment]
Hi,

it seams that the application is not able to connect to VPN and after 30 seconds it automatically stops the connection.
There is a limit on the number of devices that can connect to the VPN simultaneously. Could it be that you're currently using too many devices?

To exclude this could you disconnect from one of the devices using VPN, wait ~5 minutes and try to connect with the Linux machine?

Br,

**Comment by dbzix** [comment]
Hey, @mariusSincovici, this is not my case, as the paid plan works for 10 devices, while I have only 3.
1 mobile and 2 laptops; the mobile phone and 1st laptop work perfectly fine, but this 2nd laptop cannot connect.
The funny thing is that both laptops have the same os (Linux Mint) and the same setup of software/tools.

From what I see in the failure log, this can be an issue:
```
nordvpnd[426679]: 2024-10-03T13:09:06.758136Z ERROR libmoose::tracker::tracker: ContextNotFound Error while fetching shared context: ContextNotFound
```
because on my 1st laptop, there is no such string in the logs.

**Comment by mariusSincovici** [comment]
Hi,
that error is just something from analytics and it has no impact on the VPN functionality.
Would it be possible to try to connect using OpenVPN instead of Nordlynx? To see if there are other errors reported.
Thanks

**Comment by dbzix** [comment]
Hey, @mariusSincovici !
I already tried everything - using OpenVPN, disabling IPv6, connecting manually, etc. Do you need any logs?

**Comment by mariusSincovici** [comment]
Hi, yes please.

But if you want you could contact customer support and provide there the logs, mostly for privacy/security reasons. 
And then the information will reach to us from the customer support team. Just mention the GitHub link so we can better keep track.

**Comment by dbzix** [comment]
Hey, @mariusSincovici !
With OpenVPN I get:
```
nordvpnd[790113]: 2024/10/04 19:27:50 INFO 2024-10-04 19:27:50 Server poll timeout, restarting
```
3 times in a row, then:
```
nordvpnd[790113]: 2024/10/04 19:27:55 [Debug] notifying about disconnect event
nordvpnd[790113]: 2024/10/04 19:27:55 [Defer] not active
nordvpnd[790113]: 2024/10/04 19:27:55 [Warning] removing suppress prefix rule: no such file or directory
nordvpnd[790113]: 2024/10/04 19:27:55 [Warning] removing fwmark rule: no such file or directory
nordvpnd[790113]: 2024/10/04 19:27:55 [Warning] removing suppress prefix rule: no such file or directory
nordvpnd[790113]: 2024/10/04 19:27:55 [Warning] removing fwmark rule: no such file or directory
nordvpnd[790113]: 2024/10/04 19:27:55 [Defer] unsetting ifgroups: not set
nordvpnd[790113]: 2024/10/04 19:27:55 [Debug] notifying about disconnect event
nordvpnd[790113]: 2024/10/04 19:27:55 [Defer] not active
nordvpnd[790113]: 2024/10/04 19:27:55 [Error] failed to connect to lt16.nordvpn.com : server timeout
```

I was trying to get help with Customer Support on the site - it's just a time waste...
I'd prefer to resolve my issue here if that's possible.

This is very strange to have two identical machines, but very different behavior of nordvpn client on each of them.

**Comment by mariusSincovici** [comment]
Hi,

Ok sure we can continue the talk here.
yes, this is strange. I'll talk on Monday with the team to see if they have some suggestions of what could be the issue.

Until then, could you please provide more info about your system?
1. The OS version of Linux Mint.
2. App settings.
4. How did you've logged in? From browser or with token parameter?
5. Check the firewall just to be sure that no other program inserts some rules to block the communication with the VPN server.
6. Did it worked before version 3.19? If yes, then could you try to revert to the last working version and check?
7. Could you try to reinstall the application and restart the system?
8. If you can, check with Wireshark if there is communication with the VPN server while trying to connect(in the 30 seconds before stops the connect)?

Another quick possible test would be to check if you can use Meshnet with NordLynx and route the traffic from "problematic" system (dev2) thru the other Linux Mint(dev1). [Steps](https://meshnet.nordvpn.com/features/routing-traffic-in-meshnet/traffic-routing-linux):
1. Enable meshnet on both devices
2. execute `nordvpn meshnet peer list` in dev2 to get the dev1 peer name.
3. then in dev2 execute `nordvpn meshnet peer connect <name_of_the_dev1_peer>`
If you get and error with the 
4. Check that connection is successful and that you have internet on dev2.

Thanks

**Comment by dbzix** [comment]
Hey, @mariusSincovici !

1. Linux Mint 22.
2. Default settings, nothing was changed.
3. Via website.
4. Firewall is on default rules. Disabling it changes nothing.
5. Nope. The previous version was 3.18.3, probably? Didn't worked since the beginning.
6. Already re-installed app, cleared all the caches, rebooted system. Didn't help.
7. Will try, but need more steps to understand what to check exactly.

For the meshnet:
1. Enabled routing on both mesh nodes
2. Enabled local network access on both mesh nodes
3. Disabled firewall on both mesh nodes
Result:
```
nordvpnd[790113]: 2024/10/04 21:43:36 [Error] starting networker: context deadline exceeded
nordvpnd[790113]: 2024/10/04 21:43:41 [Warning] TELIO(v5.0.0): "telio_relay::derp":270 (DerpRelay) Failed to connect: I/O Error: Network is unreachable (os error 101)
```

By some reason `nordvpn meshnet peer list` shows main mesh node as `disconnected` while it's connected to VPN:
```
Local Peers:
Nickname: main
Hostname: xxxxx-everest.nord
Status: disconnected
```

**Comment by mariusSincovici** [comment]
Hi,
is `main` the other working Linux device? Does it mean that the devices don't see each other as connected to meshnet?

`Network is unreachable (os error 101)` would indicate that there was no internet or that the connection is blocked by firewall or some DNS issues... The application is not able to communicate with the servers.

Do you get in the logs, also for the working Linux the error: `TELIO(v5.0.0): "telio_pmtu":88 Failed to probe PMTU: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }`?

Could you please send the logs again to the customer support, to see the errors for OpenVPN connection and meshnet? We've found your ticket, so if you could replay to the existing conversation we can get the logs. 

Thanks.

**Comment by dbzix** [comment]
Hey, @mariusSincovici !
1. Yes, `main` is just the alias for the working Linux device.
2. `Network is unreachable (os error 101)` has been discovered only once. The firewall was down at this time actually.
3. Yes, I have the same message on the working machine: `TELIO(v5.0.0): "telio_pmtu":88 Failed to probe PMTU: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }`

Speaking about my experience of communicating with the customer support - they insist that my problem is because of export restrictions, though my other devices don't experience such problems... For me, that seems like an opinionated approach of 'helping' the customer. And yes, I quoted the word **helping** intentionally.

**Comment by mariusSincovici** [comment]
Hi,

Thank you for the info.

Sorry that the customer support was not able to help you solve the issue. 

The problem is that we need the latest app logs to see if there are any messages that might help us better understand the issue. Because of this I was suggesting to send them to customer support(just reply to the existing conversation)  and we can get the logs from there.
The alternative would be to put them here, but I wouldn't recommend because they will have sensitive information and can be accessed by anyone.

We would also need the Wireshark capture to see the communication between the app  and the VPN server:
1. Install wireshark and run it with sudo, e.g.: sudo wireshark
2. set the technology in NordVPN application to NordLynx
3. start recording with wireshark for your wifi interface.
4. try to connect to VPN
5. after connection fails stop Wireshark and save the file
6. export the NordVPN app log: `sudo journalctl -u nordvpnd.service > nordvpn.log`

We would need both files. 

Thanks

**Comment by dbzix** [comment]
Auto-fixed with `3.19.1`.
